### ISSUE TITLE
Checking unprotected resources with undefined path in router config

### ISSUE THREAD

**Issue by Hampfter** [AzureAD/microsoft-authentication-library-for-js#1419]
## Library
"@azure/msal-angular": "^1.0.0-beta.4"

## Framework
Angular 8

## Error message

```
core.js:4002 ERROR TypeError: Cannot read property 'indexOf' of undefined
    at azure-msal-angular.js:130
    at Array.some (<anonymous>)
    at MsalService.push../node_modules/@azure/msal-angular/fesm5/azure-msal-angular.js.MsalService.isUnprotectedResource (azure-msal-angular.js:130)
    at SafeSubscriber._next (azure-msal-angular.js:117)
    at SafeSubscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.SafeSubscriber.__tryOrUnsub (Subscriber.js:192)
    at SafeSubscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.SafeSubscriber.next (Subscriber.js:130)
    at Subscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber._next (Subscriber.js:76)
    at Subscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next (Subscriber.js:53)
    at Subject.push../node_modules/rxjs/_esm5/internal/Subject.js.Subject.next (Subject.js:47)
    at SwitchMapSubscriber.project (router.js:3832)
```

## Description
In the constructor of MsalService you`re checking the routes defined in the router configuration. The path property of a route can be undefined and you are checking it, but after you access it, so the two statments should be switched. 

```js
  _this.router.events.subscribe(function (event) {
            for (var i = 0; i < router.config.length; i++) {
                if (!router.config[i].canActivate) {
                    if (_this.msalAngularConfig.unprotectedResources) {
                        if (!_this.isUnprotectedResource(router.config[i].path) && !_this.isEmpty(router.config[i].path)) {
                            _this.msalAngularConfig.unprotectedResources.push(router.config[i].path);
                        }
                    }
                }
            }
        });
```

should be 

```js
_this.router.events.subscribe(function (event) {
            for (var i = 0; i < router.config.length; i++) {
                if (!router.config[i].canActivate) {
                    if (_this.msalAngularConfig.unprotectedResources) {
                         if (!_this.isEmpty(router.config[i].path) && !_this.isUnprotectedResource(router.config[i].path)) {
                            _this.msalAngularConfig.unprotectedResources.push(router.config[i].path);
                        }
                    }
                }
            }
        });
```

## Expected behavior
I should not get an error if my path is undefined in the router config.

**Comment by jasonnutter** [comment]
@Hampfter Thanks! I'll make sure this gets fixed.

**Comment by jasonnutter** [comment]
@Hampfter Fix in #1454
